smart-contracts/contracts/Unlock.sol

```diff
@@ -354,8 +354,11 @@ contract Unlock is
           // Get the value of 1 UDT (w/ 18 decimals) in ETH
           uint udtPrice = udtOracle.updateAndConsult(udt, 10 ** 18, weth);
 
+          // base fee is set to 100 GWEI for chains that does not support block.basefee
+          uint baseFee = block.basefee != 0 ? block.basefee : 100;
```

Comment:
I read this https://ethereum.stackexchange.com/questions/109340/error-32000-testing-block-basefee-in-bsc-testnet
I wonder if we should try/catch this? Are you sure we would only get `0` when it is not defined?

Reply:
yes makes more sense. Looking more into it, it seems to revert on BSC https://ethereum.stackexchange.com/questions/109340/error-32000-testing-block-basefee-in-bsc-testnet